CONTRIBUTING.md

```diff
@@ -39,39 +41,63 @@ terraform {
+- `make test-unit` - runs all the unit tests.
+- `make test-acc` - runs the tests with http recordings. To run a specific test pass the `FILTER` var. Usage `make test-acc FILTER="TestAccResourceServer"`.
+- `make test-acc-e2e` - runs the tests against a real Auth0 tenant. To run a specific test pass the `FILTER` var. Usage `make test-acc-e2e FILTER="TestAccResourceServer"`.
 
-To run the tests against an Auth0 tenant, use the `make test-acc-e2e` command. Start by creating an
+To run the tests against an Auth0 tenant start by creating an
 [M2M app](https://auth0.com/docs/applications/set-up-an-application/register-machine-to-machine-applications) in the
-tenant, that has been authorized to call the Management API and has all the required permissions.
+tenant, that has been authorized to request access tokens for the Management API and has all the required permissions.
 
-Then set the following environment variables:
+Then set the following environment variables on your machine:
 
 * `AUTH0_DOMAIN`: The **Domain** of the M2M app
 * `AUTH0_CLIENT_ID`: The **Client ID** of the M2M app
 * `AUTH0_CLIENT_SECRET`: The **Client Secret** of the M2M app
-* `AUTH0_DEBUG`: Set to `true` to call the Management API in debug mode, which dumps the HTTP requests and responses to
-  the output
+* `AUTH0_DEBUG`: Set to `true` to call the Management API in debug mode, which dumps the HTTP requests and responses to the output
 
-**Note:** The e2e acceptance tests make calls to a real Auth0 tenant, and create real resources. Certain tests also
-require a paid Auth0 subscription to be able to run successfully, e.g. `TestAccCustomDomain` and the ones starting with
-`TestAccLogStream*`.
+> **Note**
+> The http test recordings can be found in the [recordings](./test/data/recordings) folder.
 
-**Note:** At the time of writing, the following configuration steps are also required for the test tenant:
+> **Warning** 
+> The e2e acceptance tests make calls to a real Auth0 tenant, and create real resources. 
+> Certain tests also require a paid Auth0 subscription to be able to run successfully,
+> e.g. `TestAccCustomDomain` and the ones starting with `TestAccLogStream*`.
 
-* The `Username-Password-Authentication` connection must have _Requires Username_ option enabled for the user tests to
+> **Note** 
+> At the time of writing, the following configuration steps are also required for the test tenant:
+> - The `Username-Password-Authentication` connection must have _Requires Username_ option enabled for the user tests to
   successfully run.
 
-## Documentation
+## Adding New HTTP Test Recordings
 
-The documentation found in the [docs](./docs) folder is generated using
-[terraform-plugin-docs](https://github.com/hashicorp/terraform-plugin-docs). Please run `make docs` to regenerate
-documentation for newly added resources or schema attributes.
+To add new http test recordings or to regenerate old ones you can make use of the `make test-acc-record` command.
```

Comment:
"When creating a new Terraform resource or adding a new resource property, the HTTP recordings will need to be re-recorded."...